packages/ui/src/components/NumberInputV2/index.tsx

```diff
@@ -0,0 +1,373 @@
+  /**
+   * Text displayed into component at the right of number value.
+   */
+  unit?: string
+  tooltip?: string
+  className?: string
+  'data-testid'?: string
+  label?: string
+  /**
+   * Label description displayed right next to the label. It allows you to customize the label content.
+   */
+  labelDescription?: ReactNode
+  error?: string
+  success?: string
+  helper?: string
+} & Pick<
+  InputHTMLAttributes<HTMLInputElement>,
+  | 'onChange'
+  | 'value'
+  | 'defaultValue'
+  | 'onFocus'
+  | 'onBlur'
+  | 'name'
+  | 'id'
+  | 'placeholder'
+  | 'aria-label'
+  | 'aria-describedby'
+  | 'min'
+  | 'max'
+  | 'disabled'
+  | 'step'
+  | 'readOnly'
+  | 'required'
+  | 'autoFocus'
+>
+
+/**
+ * NumberInputV2 component is used to increment / decrement a number value by clicking on + / - buttons or
+ * by typing into input.
+ */
+export const NumberInputV2 = forwardRef(
+  (
+    {
+      disabled = false,
+      max = Infinity,
+      min = 0,
+      name,
+      onChange,
+      onFocus,
+      onBlur,
+      size = 'medium',
+      step,
+      unit,
+      value,
+      tooltip,
+      className,
+      label,
+      labelDescription,
+      id,
+      placeholder,
+      error,
+      success,
+      helper,
+      'aria-label': ariaLabel,
+      'data-testid': dataTestId,
+      required,
+      autoFocus,
+      readOnly,
+    }: NumberInputProps,
+    ref: ForwardedRef<HTMLInputElement>,
+  ) => {
+    const localRef = createRef<HTMLInputElement>()
+    useImperativeHandle(ref, () => localRef.current as HTMLInputElement)
+
+    console.log(unit, unit?.length)
+
+    const uniqueId = useId()
+    const localId = id ?? uniqueId
+
+    const onClickSideButton = useCallback(
+      (direction: 'up' | 'down') => () => {
+        if (direction === 'up') {
+          localRef.current?.stepUp()
+        }
+
+        if (direction === 'down') {
+          localRef.current?.stepDown()
+        }
+      },
+      [localRef],
+    )
+
+    const helperSentiment = useMemo(() => {
+      if (error) {
+        return 'danger'
+      }
+
+      if (success) {
+        return 'success'
+      }
+
+      return 'neutral'
+    }, [error, success])
+
+    return (
+      <Stack gap="0.5">
+        <Stack direction="row" gap="1" alignItems="center">
+          <Stack direction="row" gap="0.5" alignItems="start">
+            <Text
+              as="label"
+              variant="bodySmallStrong"
+              sentiment="neutral"
+              htmlFor={id ?? localId}
+            >
+              {label}
+            </Text>
+            {required ? <Icon name="asterisk" color="danger" size={8} /> : null}
+          </Stack>
+          {labelDescription ?? null}
+        </Stack>
+        <div>
+          <Tooltip text={tooltip}>
+            <Container
+              data-disabled={disabled}
+              data-readOnly={readOnly}
+              data-error={!!error}
+              data-success={!!success}
+              className={className}
```

Comment:
It may should be applied on the main container (Stack l274), otherwhile there is no way to apply width/position  or placement properties like that on the all component ?